### PR TITLE
interop/xds: implement new stats response field

### PR DESCRIPTION
**Pull request by dfawley**
Plus a few small go style changes & refactors.

### CODE REVIEW

**Review by menghanl — CHANGES_REQUESTED** [review]
Run the the existing tests with this to make sure it doesn't breaking anything?

File: interop/xds/client/client.go [hunk 1]
```diff
@@ -73,65 +74,75 @@ func (watcher *statsWatcher) buildResp() *testpb.LoadBalancerStatsResponse {
...
 
 type accumulatedStats struct {
 	mu                       sync.Mutex
-	numRpcsStartedByMethod   map[string]int32
-	numRpcsSucceededByMethod map[string]int32
-	numRpcsFailedByMethod    map[string]int32
+	numRPCsStartedByMethod   map[string]int32
+	numRPCsSucceededByMethod map[string]int32
+	numRPCsFailedByMethod    map[string]int32
+	RPCStatusByMethod        map[string]map[int32]int32
```
[the remark this answers is not in the document]
Comment (dfawley) [reply interop/xds/client/client.go]: Done

File: interop/xds/client/client.go [hunk 2]
```diff
@@ -73,65 +74,75 @@ func (watcher *statsWatcher) buildResp() *testpb.LoadBalancerStatsResponse {
...
 }
 
 func (as *accumulatedStats) buildResp() *testpb.LoadBalancerAccumulatedStatsResponse {
 	as.mu.Lock()
 	defer as.mu.Unlock()
 	return &testpb.LoadBalancerAccumulatedStatsResponse{
-		NumRpcsStartedByMethod:   copyStatsMap(as.numRpcsStartedByMethod),
-		NumRpcsSucceededByMethod: copyStatsMap(as.numRpcsSucceededByMethod),
-		NumRpcsFailedByMethod:    copyStatsMap(as.numRpcsFailedByMethod),
+		NumRpcsStartedByMethod:   as.numRPCsStartedByMethod,
```
[the remark this answers is not in the document]
Comment (dfawley) [reply interop/xds/client/client.go]: Aha.  Done.

File: interop/xds/client/client.go [hunk 3]
```diff
@@ -176,7 +188,7 @@ func setRPCSucceeded() {
 	atomic.StoreUint32(&rpcSucceeded, 1)
 }
 
-// Wait for the next LoadBalancerStatsRequest.GetNumRpcs to start and complete,
+// Wait for the next LoadBalancerStatsRequest.GetNumRPCs to start and complete,
```
[the remark this answers is not in the document]
Comment (dfawley) [reply interop/xds/client/client.go]: Fixed

**Review by dfawley — COMMENTED** [review]
Thanks for the review - all comments addressed.

**Review by menghanl — APPROVED** [review]

**Pull request merged**